code/modules/projectiles/ammunition.dm

```diff
@@ -47,7 +48,26 @@
 		else
 			to_chat(user, "<span class='notice'>You inscribe \"[label_text]\" into \the [initial(BB.name)].</span>")
 			BB.name = "[initial(BB.name)] (\"[label_text]\")"
-
+	else if(istype(W, /obj/item/ammo_casing))
+		if(W.type != src.type)
+			to_chat(user, SPAN_WARNING("Ammo of different types cannot stack!"))
+			return
+		if(max_stack == 1)
+			to_chat(user, SPAN_WARNING("\The [src] cannot be stacked!"))
+			return
+		if(!src.BB)
+			to_chat(user, SPAN_WARNING("That round is spent!"))
+			return
+		var/obj/item/ammo_casing/B = W
+		if(!B.BB)
+			to_chat(user, SPAN_WARNING("Your round is spent!"))
+			return
+		var/obj/item/ammo_pile/pile = new /obj/item/ammo_pile(get_turf(user))
+		pile.ammo_type = src.type
+		pile.add_ammo(src)
+		pile.add_ammo(W)
+		pile.check_name_and_ammo()
```

Comment:
You should do all this stuff in initialize, really

Reply:
initialize is reserved for mapped in things atm, rejiggering it would be a pain in the butt

Reply:
what i mean by that is, if you do this in initialize, the ammo is lost because it spawns new ammo in. i want to preserve the items because you can label bullets